src/Controls/src/Build.Tasks/CompiledMarkupExtensions/OnIdiomExtension.cs

```diff
@@ -0,0 +1,113 @@
+		memberRef = (idiomProperties.FirstOrDefault().Value as IElementNode).XmlType.GetTypeReference(context.Cache, module, node as IXmlLineInfo);
+		foreach (var idiom in idiomProperties.Skip(1))
+		{
+			var idiomTypeRef = (idiom.Value as IElementNode).XmlType.GetTypeReference(context.Cache, module, node as IXmlLineInfo);
+			if (TypeRefComparer.Instance.Compare(memberRef, idiomTypeRef) != 0)
+			{
+				//should this issue a warning?
+				memberRef = module.TypeSystem.Object;
+			}
+		}
+
+		var localName = propertyName.LocalName;
+		TypeReference declaringTypeReference = null;
+		FieldReference bpRef = null;
+		PropertyDefinition propertyRef = null;
+		if (parentNode is IElementNode && propertyName != XmlName.Empty)
+		{
+			var parentTypeRef = ((IElementNode)parentNode).XmlType.GetTypeReference(context.Cache, module, node as IXmlLineInfo);
+			bpRef = SetPropertiesVisitor.GetBindablePropertyReference(parentTypeRef, propertyName.NamespaceURI, ref localName, out _, context, node as IXmlLineInfo);
+			propertyRef = parentTypeRef.GetProperty(context.Cache, pd => pd.Name == localName, out declaringTypeReference);
+		}
+
+		var targetTypeRef = bpRef?.GetBindablePropertyType(context.Cache, node as IXmlLineInfo, module) ?? propertyRef?.PropertyType;
+
+		var instructions = new List<Instruction>();
+		
+		//var idiom = DeviceInfo.Idiom;
+		var deviceIdiomTypeRef = module.ImportReference(context.Cache, ("Microsoft.Maui.Essentials", "Microsoft.Maui.Devices", "DeviceIdiom"));
+		var idiomVarDef = new VariableDefinition(deviceIdiomTypeRef);
+		context.Body.Variables.Add(idiomVarDef);
+		instructions.Add(Create(Call, module.ImportPropertyGetterReference(context.Cache, ("Microsoft.Maui.Essentials", "Microsoft.Maui.Devices", "DeviceInfo"),
+				propertyName: "Idiom", isStatic: true)));
+		instructions.Add(Create(Stloc, idiomVarDef));
+
+		var ret_varDef = new VariableDefinition(memberRef);
+		context.Body.Variables.Add(ret_varDef);
+		
+		var ldRetValue = Create(Nop);
+		var instr_getDefault = Create(Nop);
+		
+		idioms = new[] { "Phone", "Tablet", "Desktop", "TV", "Watch" }; //avoid Default this time
+		foreach(var idiom in idioms)
+		{
+			if (node.Properties.ContainsKey(new XmlName("", idiom)))
+			{
+				//if (idiom == DeviceIdiom.Phone)
+				instructions.Add(Create(Ldloc, idiomVarDef));
+				instructions.Add(Create(Call, module.ImportPropertyGetterReference(context.Cache, ("Microsoft.Maui.Essentials", "Microsoft.Maui.Devices", "DeviceIdiom"),
+						propertyName: idiom, isStatic: true)));
+				instructions.Add(Create(Call, module.ImportMethodReference(context.Cache, ("Microsoft.Maui.Essentials", "Microsoft.Maui.Devices", "DeviceIdiom"),
+						methodName: "op_Equality",
+						parameterTypes: new[] { ("Microsoft.Maui.Essentials", "Microsoft.Maui.Devices", "DeviceIdiom"), ("Microsoft.Maui.Essentials", "Microsoft.Maui.Devices", "DeviceIdiom") },
+						isStatic: true)));
+				instructions.Add(Create(Brfalse, instr_getDefault));
```

Comment:
Is this correct? Wouldn't this jump to the `getDefault` branch if the first idiom doesn't match the idiom, effectively skipping over all the other cases?

Reply:
lemme check the generated code, and test this, but I think it's right